cloudwatch_logger/src/log_node_param_helper.cpp

```diff
@@ -0,0 +1,199 @@
+    parameter_reader->ReadBool(kNodeParamSubscribeToRosoutKey, subscribe_to_rosout);
+  switch (ret)
+  {
+    case AwsError::AWS_ERR_NOT_FOUND:
+      subscribe_to_rosout = kNodeSubscribeToRosoutDefaultValue;
+      AWS_LOGSTREAM_WARN(
+      __func__,
+      "Whether to subscribe to rosout_agg topic configuration not found, setting to default value: "
+        << kNodeSubscribeToRosoutDefaultValue);
+      break;
+    case AwsError::AWS_ERR_OK:
+      AWS_LOGSTREAM_INFO(
+      __func__, "Whether to subscribe to rosout_agg topic is set to: " << subscribe_to_rosout);
+      break;
+    default:
+      subscribe_to_rosout = kNodeSubscribeToRosoutDefaultValue;
+      AWS_LOGSTREAM_ERROR(
+        __func__,
+        "Error " << ret 
+        << "retrieving parameter for Whether to subscribe to rosout_agg topic configuration " 
+        << ", setting to default value: " << kNodeSubscribeToRosoutDefaultValue);
+  }
+  return ret;
+}
+
+Aws::AwsError ReadMinLogVerbosity(
+  std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
+  int8_t & min_log_verbosity)
+{
+  min_log_verbosity = kNodeMinLogVerbosityDefaultValue;
+
+  std::string specified_verbosity;
+  Aws::AwsError ret =
+    parameter_reader->ReadStdString(kNodeParamMinLogVerbosityKey, specified_verbosity);
+  switch (ret)
+  {
+    case AwsError::AWS_ERR_NOT_FOUND:
+      AWS_LOGSTREAM_WARN(__func__, "Log verbosity configuration not found, setting to default value: "
+                                   << kNodeMinLogVerbosityDefaultValue);
+      break;
+    case AwsError::AWS_ERR_OK:
+      if ("DEBUG" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::DEBUG;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to DEBUG.");
+      } else if ("INFO" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::INFO;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to INFO.");
+      } else if ("WARN" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::WARN;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to WARN.");
+      } else if ("ERROR" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::ERROR;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to ERROR.");
+      } else if ("FATAL" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::FATAL;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to FATAL.");
+      } else {
+        ret = AwsError::AWS_ERR_PARAM;
+        AWS_LOGSTREAM_WARN(__func__,
+                          "Log verbosity configuration not valid, setting to default value: "
+                           << kNodeMinLogVerbosityDefaultValue);
+      }
+      break;
+    default:
+      AWS_LOGSTREAM_ERROR(__func__, 
+        "Error " << ret << " retrieving log verbosity configuration, setting to default value: "
+        << kNodeMinLogVerbosityDefaultValue);
+  }
+  return ret;
+}
+
+Aws::AwsError ReadSubscriberList(
```

Comment:
Especially add comments to this, and re-order the parameter list to have parameters to be in order of how they are used, [IN] [IN/OUT][OUT].
https://google.github.io/styleguide/cppguide.html#Output_Parameters

Reply:
done